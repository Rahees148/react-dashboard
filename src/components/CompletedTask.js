import React, { useEffect, useState } from 'react';
import styled from 'styled-components';

//Context
import {useContext} from 'react';
import {TaskContext} from '../context/TaskContext';

const Card = styled.div`
  width:100%;
  padding:24px;
  background-color:#ffffff;
  box-shadow: 0px 3px 6px #0000000A;
  border-radius: 12px;
`;
const Title = styled.h3`
  color: #537178;
  font-size:20px;
  line-height:24px;
  font-weight:500;
  margin:0px 0 24px;
`;
const CompletedTaskCount = styled.h4`
  color: #5285EC;
  font-size:64px;
  line-height:78px;
  font-weight:500;
`;
const Span = styled.span`
  color: #8F9EA2;
  font-size:20px;
  line-height:24px;
`;


function CompletedTask(){
  const context = useContext(TaskContext);
  const [totalCompletedTask, setCompletedTaskTotal] = useState(0);

  useEffect(()=>{
    const tasks = context.tasks;
    const completedTask = tasks.filter(task => task.completed === true);
    setCompletedTaskTotal(completedTask.length);
  },[context]);

    return (
        <Card>
          <Title>Tasks Completed</Title>
          <CompletedTaskCount>{totalCompletedTask}<Span>/{context.tasks.length}</Span> </CompletedTaskCount>
        </Card>
    )
}

export default CompletedTask;