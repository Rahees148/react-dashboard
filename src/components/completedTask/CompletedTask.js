import React, { useEffect, useState } from 'react';
import {Style} from './CompletedTask.style';

//Context
import {useContext} from 'react';
import {TaskContext} from '../../context/TaskContext';


function CompletedTask(){
  const context = useContext(TaskContext);
  const [totalCompletedTask, setCompletedTaskTotal] = useState(0);

  useEffect(()=>{
    const tasks = context.tasks;
    const completedTask = tasks.filter(task => task.completed === true);
    setCompletedTaskTotal(completedTask.length);
  },[context]);

    return (
        <Style.Card>
          <Style.Title>Tasks Completed</Style.Title>
          <Style.CompletedTaskCount>{totalCompletedTask}
            <Style.Span>/{context.tasks.length}</Style.Span> 
          </Style.CompletedTaskCount>
        </Style.Card>
    )
}

export default CompletedTask;