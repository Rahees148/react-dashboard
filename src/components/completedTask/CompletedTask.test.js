
import renderer from 'react-test-renderer';
import CompletedTask from './CompletedTask';


it('renders correctly', () => {
  const tree = renderer
    .create(<CompletedTask />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});