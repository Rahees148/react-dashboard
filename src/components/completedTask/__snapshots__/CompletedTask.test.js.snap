// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdfBwQ efwkwq"
>
  <h3
    className="sc-hKgILt ckRRbN"
  >
    Tasks Completed
  </h3>
  <h4
    className="sc-gsTCUz hdIXul"
  >
    0
    <span
      className="sc-dlfnbm jrKrgu"
    >
      /
      0
    </span>
  </h4>
</div>
`;
